lib/puppet/provider/elasticsearch_keystore/elasticsearch_keystore.rb

```diff
@@ -107,9 +109,24 @@ def flush
       @property_flush[:settings].first.each_pair do |setting, value|
         next unless @property_hash[:settings].nil? \
           or not @property_hash[:settings].include? setting
-        debug(self.class.run_keystore(
-          ['add', '--force', '--stdin', setting], resource[:name], resource[:configdir], value
-        ))
+        if value.is_a?(Hash) and value.key?('file')
+          infile = Tempfile.new('elasticsearch-keystore')
```

Comment:
I have some doubts if keeping the key in a temp file (so probably in `/tmp`) is secure...

Reply:
It is. `Tempfile` module sets permissions to 600(http://ruby-doc.org/stdlib-2.6.3/libdoc/tempfile/rdoc/Tempfile.html#method-c-new). We would need to create temporary file anyway, because `elasticsearch-keystore` doesn't accept data from stdin. This issue is described in greater detail here: https://github.com/elastic/elasticsearch/issues/39993.

Reply:
Got it, thanks.